subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/CompositeOutputFilePropertySpec.java

```diff
@@ -20,16 +20,25 @@
 import org.gradle.internal.file.TreeType;
 import org.gradle.internal.fingerprint.OutputNormalizer;
 
+import java.io.File;
+
 public class CompositeOutputFilePropertySpec extends AbstractFilePropertySpec implements OutputFilePropertySpec {
+    private final File root;
     private final TreeType outputType;
 
-    public CompositeOutputFilePropertySpec(String propertyName, FileCollection files, TreeType outputType) {
+    public CompositeOutputFilePropertySpec(String propertyName, FileCollection files, File root, TreeType outputType) {
         super(propertyName, OutputNormalizer.class, files);
+        this.root = root;
```

Comment:
This is not a `CompositeOutputFilePropertySpec` anymore. It's perhaps a `FilteredOutputDirectoryPropertySpec`? Or at least a `FileTreeOutputPropertySpec`.

It would be great to document the difference between this thing and `DefaultCacheableOutputFilePropertySpec`.

Reply:
Renamed and documented.